optimum/intel/openvino/modeling_diffusion.py

```diff
@@ -0,0 +1,462 @@
+        return model
+
+    def reshape(
+        self,
+        batch_size: int,
+        height: int,
+        width: int,
+        num_images_per_prompt: int = -1,
+    ):
+        self.is_dynamic = -1 in {batch_size, height, width, num_images_per_prompt}
+        self.text_encoder.model = self._reshape_text_encoder(self.text_encoder.model, batch_size)
+        self.vae_decoder.model = self._reshape_vae_decoder(self.vae_decoder.model, height, width)
+        self.unet.model = self._reshape_unet(self.unet.model, batch_size, height, width, num_images_per_prompt)
+        self.clear_requests()
+        return self
+
+    def half(self):
+        """
+        Converts all the model weights to FP16 for more efficient inference on GPU.
+        """
+        compress_model_transformation(self.vae_decoder.model)
+        compress_model_transformation(self.text_encoder.model)
+        compress_model_transformation(self.unet.model)
+        self.clear_requests()
+        return self
+
+    def clear_requests(self):
+        self.text_encoder.request = None
+        self.vae_decoder.request = None
+        self.unet.request = None
+
+    def compile(self):
+        self.text_encoder._create_inference_request()
+        self.vae_decoder._create_inference_request()
+        self.unet._create_inference_request()
+
+    def __call__(self, *args, **kwargs):
+        guidance_scale = kwargs.get("guidance_scale", None)
+        if guidance_scale is not None and guidance_scale <= 1 and not self.is_dynamic:
+            raise ValueError(
+                f"`guidance_scale` was set to {guidance_scale}, static shapes are only supported for `guidance_scale` > 1, "
+                "please set `dynamic_shapes` to `True` when loading the model."
+            )
+        return StableDiffusionPipelineMixin.__call__(self, *args, **kwargs)
+
+    @classmethod
+    def _load_config(cls, config_name_or_path: Union[str, os.PathLike], **kwargs):
+        return cls.load_config(config_name_or_path, **kwargs)
+
+    def _save_config(self, save_directory):
+        self.save_config(save_directory)
+
+
+class OVModelPart:
+    def __init__(
+        self, model: openvino.runtime.Model, parent_model: OVBaseModel, ov_config: Optional[Dict[str, str]] = None
+    ):
+        self.model = model
+        self.parent_model = parent_model
+        self.input_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.inputs)}
+        self.input_dtype = {
+            inputs.get_any_name(): OV_TO_NP_TYPE[inputs.get_element_type().get_type_name()]
+            for inputs in self.model.inputs
+        }
+        self.ov_config = ov_config or self.parent_model.ov_config
+        self.request = None
+
+    def _create_inference_request(self):
+        if self.request is None:
+            logger.info("Compiling the encoder and creating the inference request ...")
+            compiled_model = core.compile_model(self.model, self.device, self.ov_config)
```

Comment:
This does not yet include model caching, which should improve model compiling time on GPU a lot. See modeling_base.py: https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/modeling_base.py#L304

Reply:
Added in [`09be7f9`](https://github.com/huggingface/optimum-intel/pull/195/commits/09be7f97c5c38ff4927f59e8e1e68b30f34c8523)